stuffer/s2n_stuffer.c

```diff
@@ -435,6 +422,35 @@ int s2n_stuffer_copy(struct s2n_stuffer *from, struct s2n_stuffer *to, const uin
     return 0;
 }
 
+int s2n_stuffer_reserve_space(struct s2n_stuffer *stuffer, uint32_t n)
+{
+    if (s2n_stuffer_space_remaining(stuffer) < n) {
+	S2N_ERROR_IF(!stuffer->growable, S2N_ERR_STUFFER_IS_FULL);
+	/* Always grow a stuffer by at least 1k */
+	const uint32_t growth = MAX(n - s2n_stuffer_space_remaining(stuffer), 1024);
+	GUARD(s2n_stuffer_resize(stuffer, stuffer->blob.size + growth));
+    }
+    return S2N_SUCCESS;
+}
+
+
+/* Copies "len" bytes from "from" to "to".
+ * If the copy cannot succeed (i.e. there are either not enough bytes available, or there is not enough space to write them
+ * restore the old value of the stuffer */
+int s2n_stuffer_copy(struct s2n_stuffer *from, struct s2n_stuffer *to, const uint32_t len)
+{
+    const uint32_t from_read_cursor = from->read_cursor;
```

Comment:
orig_read_cursor and orig_write_cursor might be a bit more clear here.

Reply:
Done